Compile error in file_reader.hpp on Windows, std::filesystem::path uses wchar_t* which cannot be passed to std::fopen expecting char*

On Windows 10, compiling with gcc on mingw / msys2, std::filesystem::path::c_str() returns a const wchar_t* while std::fopen takes a const char_t*.  This causes a compile error.

```
C:/Users/dferr/CLionProjects/PEGTL/include/tao/pegtl/internal/file_reader.hpp:29:46: error: cannot convert 'const value_type*' {aka 'const wchar_t*'} to 'const char*'
       if( auto* file = std::fopen( path.c_str(), "rb" ) )

```
You won't be able to fix it with a simple cast.  It will require a UTF-16LE aware string conversion.

I'd like to fix this and send a pull request, but it looks like the only portable way to do the encoding conversion, std::codecvt_utf8_utf16, was introduced with C++11 and _deprecated_ (without a replacement!) in C++17.

Zooming out to the larger scope, I see an alternative would be to use a wide char version of the file open function, and that's exactly what you've already did if _MSC_VER is defined (the compile error occurred in the preprocessing branch for __MINGW32__ and not _MSC_VER.  If I change the preprocessor guards so that I use the _MSC_VER of TAO_PEGTL_NAMESPACE::internal::file_open's implementation using ::_wfopen_s it does compile.

However then I get a linker error:  undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'

--- a/include/tao/pegtl/internal/file_reader.hpp
+++ b/include/tao/pegtl/internal/file_reader.hpp
@@ -26,7 +26,7 @@ namespace TAO_PEGTL_NAMESPACE::internal
       throw std::filesystem::filesystem_error( "_wfopen_s() failed", path, ec );
 #else
 #if defined( __MINGW32__ )
-      if( auto* file = std::fopen( path.c_str(), "rb" ) )
+      if( auto* file = std::fopen( path.string().c_str(), "rb" ) )
 #else
       if( auto* file = std::fopen( path.c_str(), "rbe" ) )
 #endif

